d1985r3.md

```diff
@@ -105,27 +107,34 @@ apply<H, G>;   // OK, G<int>
 
 ## "Mathematically Correct" version
 
-We need to introduce both a an _anything_ parameter (which we'll spell
-`template auto` as above), and a _wildcard_ parameter (which, given the work in
-pattern matching, we are probably free to spell `__`).
+We need to introduce an _anything_ template parameter which can bind to _any_
+template argument; we'll spell this `template auto` as above). We also need an
+_unrestricted_ template _template-parameter_ that can bind to _any_ type template;
+we will specify this by omitting the template parameter list.
 
 We can then define `apply` as follows:
 
 ```cpp
-template <template <__...> typename F, template auto... Args>
+template <template typename F, template auto... Args>
```

Comment:
I'm worried if this would be legal. We already see need for writing `typename template U<int>` where U is an UTP and we know it is bound to a class template (as just template U<int> would default to being a variable template). Thus it is scary to use the opposite keyword order for something completely different (albeit in a different part of the grammar).

Also the `template auto` part will now be ambiguous the day that variable templates are allowed as template parameters, but ok, we've all but decided to change the UTP spelling anyway.